reVX/plexos/utilities.py

```diff
@@ -0,0 +1,481 @@
+            A new set of Plexos node meta data to be merged into the meta in
+            self.
+        new_profiles : np.ndarray
+            A new set of plexos node profiles corresponding to new_meta to be
+            merged into the profiles in self where the meta data overlaps with
+            common nodes.
+        """
+
+        keep_index = []
+
+        logger.info('Merging extents with {} and {} nodes ({} total).'
+                    .format(len(self._plexos_meta), len(new_meta),
+                            len(self._plexos_meta) + len(new_meta)))
+
+        for i, ind in enumerate(new_meta.index.values):
+            lookup = (self._plexos_meta.gid.values == new_meta.loc[ind, 'gid'])
+            if any(lookup):
+                i_self = np.where(lookup)[0]
+                if len(i_self) > 1:
+                    warn('Duplicate PLEXOS node GIDs in base plexos meta!')
+                else:
+                    i_self = i_self[0]
+
+                logger.debug('Merging plexos node IDs {} and {} '
+                             '(gids {} and {})'.format(
+                                 self._plexos_meta.iloc[i_self]['plexos_id'],
+                                 new_meta.iloc[i]['plexos_id'],
+                                 self._plexos_meta.iloc[i_self]['gid'],
+                                 new_meta.iloc[i]['gid']))
+
+                self._merge_plexos_meta(self._plexos_meta, new_meta, i_self, i)
+                self._profiles[:, i_self] += new_profiles[:, i]
+            else:
+                keep_index.append(i)
+
+        new_meta = new_meta.loc[new_meta.index.values[keep_index]]
+        new_profiles = new_profiles[:, keep_index]
+
+        self._plexos_meta = pd.concat([self._plexos_meta, new_meta], axis=0,
+                                      ignore_index=True)
+        self._profiles = np.hstack((self._profiles, new_profiles))
+
+        logger.info('Merged extents. Output has {} nodes.'
+                    .format(len(self._plexos_meta)))
+
+    def merge_multiple_extents(self, meta_list, profile_list):
+        """Merge multiple plexos extents into the self attrs.
+
+        Parameters
+        ----------
+        meta_list : list
+            List of new meta data extents to merge into self.
+        profile_list : list
+            List of new gen profile to merge into self.
+
+        Returns
+        -------
+        meta : pd.DataFrame
+            Merged plexos node meta data.
+        profiles : np.ndarray
+            New profiles with merged profiles for matching nodes.
+        """
+
+        for i, meta in enumerate(meta_list):
+            self.merge_extent(meta, profile_list[i])
+
+        return self._plexos_meta, self._profiles
+
+
+class ProjectGidHandler:
+    """Class to handle project GIDs for a plexos project.
+    Can be used to make gid superset project points for 5min data."""
+
+    @staticmethod
+    def get_resource_gids(sc_table, reeds_build, wait=300,
```

Comment:
Why not just make these functions, why do they need to be in the ProjectGidHandler namespace?

Reply:
Logical grouping and thinking forward to if we want to extend this.

Reply:
great answer